cargo-geiger/src/mapping/geiger.rs

```diff
@@ -0,0 +1,148 @@
+use super::ToCargoGeigerSource;
+
+use crate::mapping::ToCargoMetadataPackage;
+
+use cargo_metadata::Metadata;
+use url::Url;
+
+impl ToCargoGeigerSource for cargo_metadata::PackageId {
+    fn to_cargo_geiger_source(
+        &self,
+        metadata: &Metadata,
+    ) -> cargo_geiger_serde::Source {
+        eprintln!("PackageId Repr: {}", self.repr);
+        let package = self.to_cargo_metadata_package(metadata).unwrap();
+
+        match package.source {
+            Some(source) => handle_source_repr(&source.repr),
+            None => handle_path_source(self),
+        }
+    }
+}
+
+fn handle_source_repr(source_repr: &str) -> cargo_geiger_serde::Source {
+    let mut source_repr_vec = source_repr.split('+').collect::<Vec<&str>>();
+
+    let source_type = source_repr_vec[0];
+
+    match source_type {
+        "registry" => {
+            cargo_geiger_serde::Source::Registry {
+                // It looks like cargo metadata drops this information
+                name: String::from("crates.io"),
```

Comment:
I'm unsure on this, it looks like it gets dropped by cargo_metadata, I'm not sure if we want to parse this from somewhere else (or even where we would parse it from)?

Reply:
I think the comment is good enough for now.